homeassistant/components/zimi/coordinator.py

```diff
@@ -0,0 +1,96 @@
+"""Zimi Controller coordinator class device."""
+
+import logging
+import pprint
+
+from zcc import ControlPoint, ControlPointDescription, ControlPointError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_TIMEOUT, CONF_VERBOSITY, CONF_WATCHDOG, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZimiCoordinator(DataUpdateCoordinator):
+    """Coordinates a single Zimi Controller hub.
+
+    Initial list of devices is fetched in _async_setup() but all subsequent data
+    is pushed from the zcc helper and handled in the Entities.
+    """
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=entry,
+            name="Zimi Cloud Connect",
+            always_update=False,
+        )
+
+        if self.config_entry:
+            _LOGGER.debug("Initialising:\n%s", pprint.pformat(self.config_entry.data))
+
+        self.api: ControlPoint | None = None
+
+    async def _async_setup(self):
+        """Set up the coordinator."""
+
+        try:
+            _LOGGER.info(
+                "Connecting to %s:%d with verbosity=%s, timeout=%d and watchdog=%d",
+                self.config_entry.data[CONF_HOST],
+                self.config_entry.data[CONF_PORT],
+                self.config_entry.data[CONF_VERBOSITY],
+                self.config_entry.data[CONF_TIMEOUT],
+                self.config_entry.data[CONF_WATCHDOG],
+            )
+
+            self.api = ControlPoint(
+                description=ControlPointDescription(
+                    host=self.config_entry.data[CONF_HOST],
+                    port=self.config_entry.data[CONF_PORT],
+                ),
+                verbosity=self.config_entry.data[CONF_VERBOSITY],
+                timeout=self.config_entry.data[CONF_TIMEOUT],
+            )
+
+            await self.api.connect()
+            _LOGGER.info("Connected")
```

Comment:
info is reserved, use `debug` please

Reply:
Done.   https://github.com/markhannon/core-integration-zimi/commit/7a83f768a7ed1d1027d342da58f7be5516167b87